File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.py
class Solution(object):
    def findLucky(self, arr):
        freq={}
        ar=[-1]
        for i in range(len(arr)):
            val=arr[i]
            if val in freq:
                freq[val]+=1
            else:
                freq[val]=1
        for key in freq:
            if key==freq[key]:
                ar.append(freq[key])
        return max(ar)
        
        

File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.js
var findLucky = function(arr) {
    let freq={}
    let ar=[-1]
    for (i=0;i<arr.length;i++){
        val=arr[i]
        if(freq[val]){
            freq[val]++
        }else{
            freq[val]=1
        }

    }
    for (let key in freq){
        if (key==freq[key]){
            ar.push(freq[key])
        }
    }
    return Math.max(...ar)
    
    
};